src/Tickets/Commerce/Gateways/Contracts/Traits/Paid_Gateway.php

```diff
@@ -12,15 +12,25 @@
 use TEC\Tickets\Commerce\Cart;
 
 trait Paid_Gateway {
+
 	/**
-	 * @inheritDoc
+	 * Determine whether the gateway should be shown as an available gateway.
+	 *
+	 * @since 5.1.6
+	 * @since TBD Prevented the cart totals from being recalculated multiple times.
+	 *
+	 * @return bool Whether the gateway should be shown as an available gateway.
 	 */
 	public static function should_show() {
+		static $cart_total = null;
```

Comment:
Avoid any type of cache we can't clear. I know this only for the request's run cycle but prefer using tribe_cache instead. If for some reason, someone calls wp_cache_flush its expected that every cache will clear. The same should happen for run time cache :D

Reply:
I am a bit confused why are we caching values of `get_cart_total`? We already have this similarly cached in the [method](https://github.com/the-events-calendar/event-tickets/blob/6ef2fe23d8b80f86707b410ee97e194278061fb7/src/Tickets/Commerce/Cart/Abstract_Cart.php#L81) itself. We may improve that variable cache as Dimi suggested but if that cache is in place we don't need caches in other places for this call.